src/core/shaderlib/project/viewport-uniforms.js

```diff
@@ -153,7 +159,8 @@ function calculateViewportUniforms({viewport, coordinateSystem, coordinateOrigin
   const distanceScales = viewport.getDistanceScales();
 
   // TODO - does this depend on useDevicePixels?
-  const devicePixelRatio = (typeof window !== 'undefined' && window.devicePixelRatio) || 1;
+  const devicePixelRatio =
+    useDevicePixels && typeof window !== 'undefined' ? window.devicePixelRatio : 1;
```

Comment:
Why not just pass the device pixel ratio? I think we have this logic in quite a few places.